flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/taskmanager/ThreadDumpInfo.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages.taskmanager;
+
+import org.apache.flink.runtime.rest.messages.ResponseBody;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Objects;
+
+/**
+ * Class containing thread dump information.
+ */
+public class ThreadDumpInfo implements ResponseBody, Serializable {
+	private static final long serialVersionUID = 1L;
+
+	public static final String FIELD_NAME_THREAD_INFOS = "threadInfos";
+
+	@JsonProperty(FIELD_NAME_THREAD_INFOS)
+	private final Collection<ThreadInfo> threadInfos;
```

Comment:
Why not a map thread_name -> thread_info? I think this would be more generally usable.

Reply:
The main reason was the duplicated thread name information as a key and as part of the `ThreadInfo` class. But I guess this is premature optimization. I will change it.

Reply:
Actually, I think whether it is stored in a map or not is an implementation detail. Moreover, we don't offer a method `getThreadInfo(String threadName)` yet. Hence, there is actually no reason to store them internally as a map. Once the requirement changes, I'm happy to adapt the class.

Reply:
I wasn't talking about changing the internal storage, but to expose it as a map instead (i.e., changing the JOSN representation)

Assuming a user is interested in a particular thread, then with a map one can do `threadDump['myThreadName']`, instead of having to iterate over the array.

Reply:
Good point. I will take another look.

Reply:
After having looked at it again, I still believe that the response should be a collection of `ThreadInfos`. If the user needs to do a named lookup, then she can still create a map from the collection. It should not be the responsibility response class to present the elements in a certain order. Moreover, we don't have this requirement at the moment. Hence, I won't do the proposed change.